indent/ruby.vim

```diff
@@ -549,6 +569,14 @@ function! s:AfterIndentKeyword(pline_info)
     " level
     if s:Match(info.plnum, s:end_end_regex)
       let ind = indent('.')
+    elseif s:IsAssignment(info.pline, col)
+      if g:ruby_indent_end_alignment == 'hanging'
+        " hanging indent
+        let ind = col + info.sw - 1
+      else
+        " align with variable
+        let ind = indent(info.plnum) + info.sw
+      endif
```

Comment:
Right, I think I get it, this one is for the `something_else` line, and the `elsif` line is handled by the `DeindentingKeyword` callback. Both seem to be necessary to correctly indent the whole expression.